test/lib/parse/promise.js

```diff
@@ -39,6 +39,24 @@ describe('parsePromise', () => {
     })
   })
 
+  context('when the promise text contains a comparator keyword and a military time', () => {
+    def('urtext', () => '/do-the-thing-before-leaving-work-by-1800')
+
+    it('correctly parses the deadline from the promise text', () => {
+      const promise = {
+        id: 'testuser/do-the-thing-before-leaving-work-by-1800',
+        slug: 'do-the-thing-before-leaving-work',
+        timezone: 'etc/UTC',
+        what: 'Do the thing before leaving work',
+        urtext: 'do-the-thing-before-leaving-work-by-1800',
+      }
+      const tdue = moment($parsedPromise.tdue).toISOString()
+
+      expect($parsedPromise).to.include(promise)
+      expect(tdue).to.eq(moment.utc({ hour:18 }).toISOString())
```

Comment:
nice, much cleaner than what i had

(one little nit⛏... can we add a space after the colon in the object property - e.g. `hour: 18` - not sure why codeclimate isn't catching that with eslint)